docs/architecture-decisions/0004-metadata-json-spec.md

```diff
@@ -0,0 +1,51 @@
+# 1. Submission message spec
+
+Date: 2021-08-30
+
+## Status
+
+Proposed
+
+## Context
+
+Multiple external applications will submit DSpace item-level metadata as JSON
+files that need to be processed by this system, so we need a specification they
+can follow to create consistent metadata files with a known structure for this
+application to handle..
+
+## Decision
+
+We will use the following submission metadata file structure:
+
+Each Item's metadata (which will be identified in the message body
+"MetadataLocation" field as specified in adr-0003) will be provided in a single
+JSON file that consists of an array of objects, each object representing a
+single field in DSpace. It is up to the submitting system to ensure that the
+field names are accurate and the field values are compliant with DSpace
+requirements, this system will simply parse and post all of the fields as
+provided.
+
+The metadata file should be structured as follows:
+
+```
+{
+  "metadata": [
+    {
+      "key": "<Name of the DSpace metadata field in qualified Dublin Core format, e.g. 'dc.title'",
+      "value": "<String value of the field, e.g. 'A Very Important Thesis'",
+      "language": "<Optional - language of the field's value>"
```

Comment:
We likely want to link to an agreed upon standard for language codes.

Reply:
(or remove this if we don't have an immediate use for it... I doubt ETD can be 100% sure of the language for any fields it would be useful for can we?)

Reply:
No but the submitting systems can (and should be able to) provide languages for the fields. Just like the other parts of the metadata, it's up to the submitting applications to provide correct language codes...we'll just pass them through and let DSpace handle it.

Reply:
I think there are two things going on here:
1) I want us to define where the valid language codes come from. If not in this ADR, then someplace where consumers of this application (me!) will know what is a valid language code.
2) yes, I'm fine with consumers of this app being charged with defining which language each field is... but my assumption is that even those will be guesses. Do we know if _all_ MIT theses abstracts (as one example) are required to be in English and what do we do if they are not? (Not what does this system do... it doesn't care... but what does ETD do if the official abstract is in non-English? We don't check it. We don't have a way for a processor (or student) to state the language. I'm curious if we'll ever have data in which we'll _know_ what the language is in a way that won't negatively impact the data if we guess wrong and thus I'm wary to use this and am looking for more info as to how we can help future us to know when / if this will ever be a good idea to include. This is quite likely beyond the usefulness of discussing in a PR and I'll state that having this field optional is 100% fine, I just want to chat more about it offline at some point :)

Setting part 2 aside as possibly me over thinking things, I'd very much like us to document the expectations our target system has for language codes to not perpetuate the rampant "eng, fre, ita" vs "English", "French", "Italian" vs "eng", "fre", "ita" we see in library metadata so often. We don't need to validate this data, but I'd like to know where to go to find it.

Reply:
So...there are no valid (or invalid) language codes. I just successfully posted an item where the title field had language "gobbledygook". Which means that just like the value of each metadata field, it's up to our stakeholders (primarily Sadie) to tell us what the language should be and which fields should include it. I'll create a ticket to confirm that with her for ETD, and Wiley will need to do the same.